Userland/Libraries/LibJS/Runtime/PromisePrototype.cpp

```diff
@@ -112,21 +111,23 @@ JS_DEFINE_NATIVE_FUNCTION(PromisePrototype::finally)
             auto* promise = TRY(promise_resolve(global_object, constructor, result));
 
             // iii. Let returnValue be a new Abstract Closure with no parameters that captures value and performs the following steps when called:
-            // iv. Let valueThunk be ! CreateBuiltinFunction(returnValue, 0, "", « »).
-            auto* value_thunk = NativeFunction::create(global_object, "", [value](auto&, auto&) -> ThrowCompletionOr<Value> {
+            auto return_value = [value](auto&, auto&) -> ThrowCompletionOr<Value> {
```

Comment:
(I know it is not completely related to this PR so feel free to ignore it but) Should this `value` not be in a handle?
Also a similar question for `reason` below (line 142)

Reply:
Great catch! Added another commit to fix this.